dataflow/src/main/java/org/checkerframework/dataflow/util/UnmodifiableIdentityHashMap.java

```diff
@@ -0,0 +1,191 @@
+package org.checkerframework.dataflow.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A wrapper around an {@link IdentityHashMap} that makes it unmodifiable. All mutating operations
+ * throw {@link UnsupportedOperationException}, and all other operations delegating to the
+ * underlying map.
+ *
+ * <p>This class extends {@link IdentityHashMap} only so it is assignable to variables / fields of
+ * static type {@link IdentityHashMap}. All valid operations are delegated to the wrapped map, and
+ * any inherited state from the superclass is unused.
+ */
+@SuppressWarnings("nullness")
```

Comment:
I tried to add nullness annotations to this type but got confused.  Here are a couple of the CI errors I got:
```
/__w/1/s/dataflow/src/main/java/org/checkerframework/dataflow/util/UnmodifiableIdentityHashMap.java:71: error: [keyfor:contracts.conditional.postcondition] conditional postcondition is not satisfied when containsKey returns true.
    return map.containsKey(key);
    ^
  found   : key is @KeyFor("this.map")
  required: key is @KeyFor("this")
/__w/1/s/dataflow/src/main/java/org/checkerframework/dataflow/util/UnmodifiableIdentityHashMap.java:101: error: [nullness:override.receiver] Incompatible receiver type
  public boolean equals(@Nullable Object o) {
                 ^
  found   : @Initialized @NonNull UnmodifiableIdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object>
  required: @Initialized @Nullable IdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object>
  Consequence: method in @Initialized @NonNull UnmodifiableIdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object>
    @Initialized @NonNull boolean equals(@Initialized @NonNull UnmodifiableIdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object> this, @Initialized @Nullable Object p0)
  cannot override method in @Initialized @NonNull IdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object>
    @Initialized @NonNull boolean equals(@Initialized @Nullable IdentityHashMap<K extends @Initialized @Nullable Object, V extends @Initialized @Nullable Object> this, @Initialized @NonNull Object p0)
/__w/1/s/dataflow/src/main/java/org/checkerframework/dataflow/util/UnmodifiableIdentityHashMap.java:102: error: [nullness:argument] incompatible argument for parameter o of equals.
```
Here is the build:
https://dev.azure.com/checkerframework/checkerframework/_build/results?buildId=8760&view=logs&jobId=794427f4-26d8-57f0-fc22-31159cf0c9b8&j=794427f4-26d8-57f0-fc22-31159cf0c9b8&t=83f7d39a-b612-5a99-9aa3-cd2517324f7f

If we want to fully annotate this wrapper type for nullness, I could use some help.  Otherwise, we can add a comment to the suppression.